### PR TITLE
fix errors from testing account service setup

### PR DESCRIPTION
**Pull request by JoeCap08055**
# Description
In the process of testing the latest version of this package in the chain-setup for [account-service](https://github.com/AmplicaLabs/account-service), some bugs were found. This PR does the following:

- More readable error message when failing to fund an account properly in `UserBuilder.build()`
- In `transactions.ts`, when looking for completed transactions we were subscribing to all chain events via the system RPC, which could get events from extrinsics not submitted by us. Instead, process the events inside the transaction subscription itself.
- `schema_id` & `target_hash` had been hard-coded when provisioning graph public keys; instead, use the detected values
- Make `EventError` more readable

### CODE REVIEW

**Review by mattheworris — APPROVED** [review]
- Read through changes
🚢 it!

File: src/scaffolding/graph.ts [hunk 1]
```diff
@@ -71,8 +72,8 @@ export async function getAddGraphKeyPayload(
   ];
 
   const graphKeyAction: any = {
-    targetHash: 0,
-    schemaId: 7,
+    targetHash,
+    schemaId: publicGraphKeySchema.id,
```
Comment (claireolmstead) [inline src/scaffolding/graph.ts:76]: Does this not having a default cause any issues if no value is passed? Id there ever a case where no value is passed?

**Review by claireolmstead — APPROVED** [review]
Lots of good type fixes. 🚢

File: src/scaffolding/graph.ts (hunk 1 above)
Comment (JoeCap08055) [reply src/scaffolding/graph.ts:76]: I mean... the only gotcha is that you must call `fetchPublicKeySchema` at some point first before calling this method. Might be worth mentioning in comments. We could use put in a fall-back default value, but I think it's better to have this fail. We're supposed to resolve the schema on-chain first, and if we don't all bets should be off.

**Review by saraswatpuneet — APPROVED** [review]
Lgtm

**Pull request merged**